core/src/test/resources/sql/winagg.iq

```diff
@@ -78,6 +78,87 @@ from emp;
 
 !ok
 
+# [CALCITE-6538] OVER (ROWS CURRENT ROW) should return a window with one row, not all rows
+# (RANGE CURRENT ROW returns all rows, and is already correct.)
+select ename,
+  sal,
+  sum(sal) over (rows current row) as row_sum_sal,
+  sum(sal) over (range current row) as range_sum_sal
+from emp
+where job = 'MANAGER';
++-------+---------+-------------+---------------+
+| ENAME | SAL     | ROW_SUM_SAL | RANGE_SUM_SAL |
++-------+---------+-------------+---------------+
+| BLAKE | 2850.00 |     2850.00 |       8275.00 |
+| CLARK | 2450.00 |     2450.00 |       8275.00 |
+| JONES | 2975.00 |     2975.00 |       8275.00 |
++-------+---------+-------------+---------------+
+(3 rows)
+
+!ok
+
+# Other windows with ROWS and no ORDER BY
+select count(*) over (rows between 3 preceding and 2 preceding) as c3p2p,
+  sum(two) over (rows between 3 preceding and 2 preceding) as s3p2p
+from (select *, 2 as two from emp)
+order by 1;
++-------+-------+
+| C3P2P | S3P2P |
++-------+-------+
+|     0 |       |
+|     0 |       |
+|     1 |     2 |
+|     2 |     4 |
+|     2 |     4 |
+|     2 |     4 |
+|     2 |     4 |
+|     2 |     4 |
+|     2 |     4 |
+|     2 |     4 |
+|     2 |     4 |
+|     2 |     4 |
+|     2 |     4 |
+|     2 |     4 |
++-------+-------+
+(14 rows)
+
+!ok
+
+!if (false) {
+# The result below is from Postgres.
+# Calcite currently gives the wrong results for s1p (first 2 rows),
```

Comment:
do you know if there is an issue for these cases?

Reply:
There is a problem, but there is no separate jira case. In my first commit I wanted to avoid the issue. In my 2nd/3rd commits I addressed it and enabled the test.

So I consider that CALCITE-5883 (and this fix) covers all of these related problems.

Reply:
I asked a similar question on the mailing list a while ago, I wonder whether that test will work now too:
https://lists.apache.org/thread/jhzxf7k7wxgwv7szp5lj9mkocjq3n5pf